src/runtime/omp/omp_queue.cpp

```diff
@@ -169,25 +185,67 @@ class omp_instrumentation_setup {
   std::shared_ptr<omp_execution_finish_timestamp> _finish;
 };
 
-}
+#ifdef HIPSYCL_WITH_SSCP_COMPILER
+struct work_group_info {
+  work_group_info(rt::range<3> num_groups, rt::id<3> group_id,
+                  rt::range<3> local_size, void *local_memory)
+      : _num_groups(num_groups), _group_id(group_id), _local_size(local_size),
+        _local_memory(local_memory) {}
+
+  rt::range<3> _num_groups;
+  rt::range<3> _group_id;
+  rt::range<3> _local_size;
+  void *_local_memory;
+};
 
+using omp_sscp_kernel = void(const work_group_info *, void **);
 
-omp_queue::omp_queue(backend_id id)
-: _backend_id(id) {}
+result launch_kernel_from_so(void *handle, const std::string &kernel_name,
+                             const rt::range<3> &num_groups,
+                             const rt::range<3> &local_size,
+                             unsigned shared_memory, void **kernel_args) {
+  auto kernel = (omp_sscp_kernel *)detail::get_symbol_from_library(
+      handle, kernel_name, "omp_sscp_exectuable_object");
+
+  if (!kernel) {
+    return make_error(
+        __hipsycl_here(),
+        error_info{"omp_queue: Could not load kernel from shared object"});
+  }
 
-omp_queue::~omp_queue() {
-  _worker.halt();
+#ifdef _OPENMP
+#pragma omp parallel for collapse(3)
+#endif
+  for (std::size_t k = 0; k < num_groups.get(2); ++k) {
+    for (std::size_t j = 0; j < num_groups.get(1); ++j) {
+      for (std::size_t i = 0; i < num_groups.get(0); ++i) {
+        sycl::detail::host_local_memory::request_from_threadprivate_pool(
+            shared_memory);
+        work_group_info info{
+            num_groups, rt::id<3>{i, j, k}, local_size,
+            hipsycl::sycl::detail::host_local_memory::get_ptr()};
+        kernel(&info, kernel_args);
```

Comment:
Not something we need to solve in this PR, but something to keep in mind: Passing in `void**` might limit llvm-to-backend's capability to perform optimizations like dead-argument-elimination. I think hardwiring arguments to constants might still work if that optimization is run prior to backend flavoring in `llvm-to-host`...

Reply:
Hmm... I would assume that DAE should happen rather early on, before any backend flavoring..?

In the end, we need to find a way to pass the arguments to the kernel, and since we don't have compile-time information about this, we need some runtime "variadic" way to do this. the `void**` thing seems pretty simple.. but haven't put much thought into alternatives, yet tbh. :)

Reply:
Ideally DAE happens after the optimization pipeline (without optimizations, we might not easily know whether an argument is actually used), which has to be run after backend flavoring. If we ran flavoring after optimizations, we wouldn't optimize the bitcode libraries that we link.
But I can see how we need some kind of type erasure in order to be able to define a function pointer in the runtime.

Reply:
Perhaps one solution might be to introduce another step between `prepareIR()` which does flavoring and optimization, and `translateToBackendFormat`. We could then move the the generation of the wrapper kernel interface to that new stage.